util/design/README.md

```diff
@@ -0,0 +1,179 @@
+
+The following snippet shows a memory image configuration that puts the device into the `DEV` life cycle state, and that defines the SECRET2 partition items and locks down the partition:
+```
+{
+    // Seed to be used for generation of partition randomized values.
+    // Can be overridden on the command line.
+    seed: 01931961561863975174
+
+    // The partition and item names must correspond with the OTP memory map.
+    partitions: [
+        {
+            name:  "SECRET2",
+            // If True, a digest will be calculated for this partition.
+            // Note that this only applies to partitions with a hardware digest.
+            lock:  "True",
+            items: [
+                {
+                    name:  "RMA_TOKEN",
+                    // This item is assigned is a fixed Hex value
+                    value: "0x9cfbd7383959a62a4438bc468d76eed6 ",
+                }
+                {
+                    name:  "CREATOR_ROOT_KEY_SHARE0",
+                    // This item will be assigned a random value, based on the seed above.
+                    value: "<random>",
+                }
+                {
+                    name:  "CREATOR_ROOT_KEY_SHARE1",
+                    // This item will be assigned a random value, based on the seed above.
+                    value: "<random>",
+                }
+            ],
+        }
+        {
+            name:  "LIFE_CYCLE",
+            // Can be one of the following strings:
+            // RAW, TEST_UNLOCKED0-3, TEST_LOCKED0-2, DEV, PROD, PROD_END, RMA, SCRAP
+            state: "DEV",
+            // Can be either "RAW", or any number from 1 to 16.
+            count: "5"
+        }
+    ]
+}
+```
+
+Common example configuration files that can be used for simulation and emulation are checked in under `hw/ip/otp_ctrl/data`, e.g. `hw/ip/otp_ctrl/data/otp_ctrl_img_dev.hjson` which provisions all buffered partitions and puts the device into the `DEV` life cycle.
+
+Note that the preload image generator script automatically scrambles secret partitions, computes digests of locked partitions using the PRESENT cipher, and computes the OTP ECC bits.
+
+The OTP preload image generator expects at least one main image configuration file to be specified with the `--img-cfg` switch, for example:
+```console
+$ cd ${PROJ_ROOT}
+$ ./util/design/gen-otp-img.py --img-cfg hw/ip/otp_ctrl/data/otp_ctrl_img_dev.hjson \
+                               --out otp-img.mem
+```
+
+Additional configuration files can be added using the `--add-cfg` switch.
+The switch can be specified multiple times, and the configuration files are processed in the order they are specified.
+This allows to incrementally "patch in" additional data, if needed.
+
+For example, this can be used to patch in additional software configuration data, specified in an additional file `otp_ctrl_img_sw_cfg.hjson` that looks as follows (it is also possible to split the individual items into their own files):
```

Comment:
CC @gkelly @mcy 

The current plan is that this script is run prior to simulation or FPGA synthesis as part of the FuseSoC or Dvsim flow in order to assemble the OTP memory image.
The script populates the life cycle state with the correct encoding and takes care of scrambling secret partitions, calculating digests, adding ECC, etc before writing the image as a hexdump that can be read by the EDA tools.

Additional data (such as the software configuration partition blobs) can be patched into the image as described here in this readme.

Does that sound reasonable to you? 
Let me know if you have any thoughts or suggestions...